lib/signs_ui_web/controllers/messages_controller.ex

```diff
@@ -177,4 +211,39 @@ defmodule SignsUiWeb.MessagesController do
       _ -> {:error, "invalid expiration"}
     end
   end
+
+  defp paginate_text(text, max_length \\ 24) do
+    String.split(text)
+    |> Stream.chunk_while(
+      nil,
+      fn word, acc ->
+        if is_nil(acc) do
+          {:cont, word}
+        else
+          new_acc = acc <> " " <> word
+
+          if String.length(new_acc) > max_length do
+            {:cont, acc, word}
+          else
+            {:cont, new_acc}
+          end
+        end
```

Comment:
Could this chunk be simplified a bit by doing something like this instead?

```
new_acc =
  if is_nil(acc),
    do: word,
    else: acc <> " " <> word

if String.length(new_acc) > max_length,
  do: {:cont, acc, word},
  else: {:cont, new_acc}

Reply:
That would change the semantics in the case where a single word was longer than the limit, and cause it to emit a `nil`, so that won't work.